live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/util/E2ETestBase.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.util;
+
+import static lombok.AccessLevel.PROTECTED;
+
+import static org.alfresco.hxi_connector.live_ingester.util.ContainerSupport.HX_INSIGHT_INGEST_ENDPOINT;
+import static org.alfresco.hxi_connector.live_ingester.util.ContainerSupport.REPO_EVENT_TOPIC;
+
+import java.time.Duration;
+import jakarta.jms.Connection;
+import jakarta.jms.ConnectionFactory;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import lombok.NoArgsConstructor;
+import lombok.SneakyThrows;
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import org.wiremock.integrations.testcontainers.WireMockContainer;
+
+@SpringBootTest(properties = {
+        "logging.level.org.alfresco=DEBUG"
+})
+@Testcontainers
+@DirtiesContext // Forces framework to kill application after tests (i.e. before testcontainers die).
+@NoArgsConstructor(access = PROTECTED)
+public class E2ETestBase
```

Comment:
This probably could be `abstract` as it's not intended to contain/run any test cases, WDYT?

Reply:
I initially had it as abstract, but PMD pointed out that it was neater to have a protected constructor to prevent accidental instantiation. I don't think it really matters though as no one is going to try to create one of these.

Reply:
Oh, got it... I'm fine with this approach